server/partition.go

```diff
@@ -95,23 +95,35 @@ type partition struct {
 //
 // A partitioned stream maps to separate NATS subjects: subject, subject.1,
 // subject.2, etc.
-func (s *Server) newPartition(protoPartition *proto.Partition, recovered bool) (*partition, error) {
+func (s *Server) newPartition(protoPartition *proto.Partition, recovered bool, protoStreamsConfig *proto.CustomStreamsConfig) (*partition, error) {
```

Comment:
Make sure to at least document the flow of `CreateStreamRequest` to ensure that we do not miss anything .

In short, we have to go through all `newPartition` 's caller to ensure the changes have been made , as the signature of this function has now been changed